src/wp-includes/post.php

```diff
@@ -6460,6 +6462,19 @@ function wp_delete_attachment_files( $post_id, $meta, $backup_sizes, $file ) {
 					$deleted = false;
 				}
 			}
+			// Check for alternate size mime types in the sizeinfo['sources'] array to delete.
+			if ( isset( $sizeinfo['sources'] ) && is_array( $sizeinfo['sources'] ) && $primary_mime ) {
+				foreach ( $sizeinfo['sources'] as $mime => $properties ) {
+					if ( $mime === $primary_mime || ! is_array( $properties ) || empty( $properties['file'] ) ) {
+						continue;
+					}
+
+					$intermediate_file = str_replace( wp_basename( $file ), $properties['file'], $file );
+					if ( ! wp_delete_file_from_directory( $intermediate_file, $intermediate_dir ) ) {
+						$deleted = false;
+					}
+				}
+			}
```

Comment:
Maybe we can simplify the logic here: I don't think we need to keep track of the `$primary_mime` and compare with it here, we can simply delete all files from `$sizeinfo['sources']`. We only need to delete the file from `$sizeinfo` if there are no `$sizeinfo['sources']` at all (since the sources include whatever is stored directly within `$sizeinfo`). In other words, more like:
```php
if ( isset( $sizeinfo['sources'] ) && is_array( $sizeinfo['sources'] ) ) {
	foreach ( $sizeinfo['sources'] as $mime => $properties ) {
		// Delete file from $properties['file'].
	}
} else {
	// Delete file from $sizeinfo['file'].
}

Reply:
updated in current code